### PR TITLE
Enable new SD agents by default

### PR DESCRIPTION
**Pull request by lluunn**
Now that we have k8s 1.10.

/cc @kunmingg

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1252)

### CODE REVIEW

**Comment by jlewi** [comment]
@lluunn You probably need to rebase to get the tests fixed.

/assign @kunmingg

File: scripts/gke/deployment_manager_configs/cluster.jinja [hunk 1]
```diff
@@ -66,20 +66,10 @@ resources:
       # Create a very small minimal pool. Actual nodes will be managed
       # as additional node pools. This makes it easier to
       initialNodeCount: {{ properties['cpu-pool-initialNodeCount'] }}
-      {% if properties['stackdriver-kubernetes'] %}
-      # TODO: remove alpha when 10.2 is public.
-      # https://github.com/kubeflow/kubeflow/issues/821
-      enableKubernetesAlpha: true
       # We need 1.10.2 to support Stackdrivier GKE.
       initialClusterVersion: 1.10.2-gke.0
```
Comment (kunmingg) [inline scripts/gke/deployment_manager_configs/cluster.jinja]: Now most recent version is 1.10.5-gke.3, and 1.10.2-gke.0 is no long available from UI.

File: scripts/gke/deploy.sh [hunk 2]
```diff
@@ -159,6 +159,10 @@ if ${KUBEFLOW_DEPLOY}; then
 
   # Install the GPU driver. It has no effect on non-GPU nodes.
   kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/stable/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+
+  # Install Stackdriver Kubernetes agents.
```
Comment (jlewi) [inline scripts/gke/deploy.sh]: Why do we need to do this? Aren't the stackdriver agents installed by default?

Comment (jlewi) [reply scripts/gke/deploy.sh]: The beta instructions don't mention installing these agents
https://cloud.google.com/monitoring/kubernetes-engine/installing?

File: scripts/gke/deployment_manager_configs/cluster.jinja [hunk 3]
```diff
@@ -66,20 +66,10 @@ resources:
...
-      # Logging and monitoring have default value [logging/monitoring].googleapis.com
-      # if not set. We are using the new Stackdricer Kubernetes agents here.
-      # See cloud.google.com/monitoring/kubernetes-engine/.
-      loggingService: logging.googleapis.com/kubernetes
-      monitoringService: monitoring.googleapis.com/kubernetes
-      {% else %}
-      initialClusterVersion: 1.9.6-gke.1
-      {% endif %}
+      initialClusterVersion: 1.10.5-gke.3
+      loggingService: none
```
Comment (jlewi) [inline scripts/gke/deployment_manager_configs/cluster.jinja]: Why don't we need to  set loggingService and monitoringService anymore?

File: scripts/gke/deploy.sh (hunk 2 above)
Comment (lluunn) [reply scripts/gke/deploy.sh]: The instruction there are using gcloud or pantheon. We are doing manually here.

File: scripts/gke/deployment_manager_configs/cluster.jinja (hunk 3 above)
Comment (lluunn) [reply scripts/gke/deployment_manager_configs/cluster.jinja]: Those are legacy ones.

**Comment by lluunn** [comment]
/hold
I might be wrong.

File: scripts/gke/deploy.sh (hunk 2 above)
Comment (jlewi) [reply scripts/gke/deploy.sh]: Should we take advantage of the YAML support in ksonnet and start a package for all the YAML specs for GKE?

We could do this in a follow on a PR. I think the question is how can we make it easier to keep in sync between our web app and deploy.sh

Right now we'd have to specify the YAML spec in two different locations.

Comment (lluunn) [reply scripts/gke/deploy.sh]: https://github.com/kubeflow/kubeflow/issues/1288

**Pull request merged**